common/persistence/src/main/java/app/coronawarn/server/common/persistence/repository/FederationBatchInfoRepository.java

```diff
@@ -20,29 +20,37 @@
 
 package app.coronawarn.server.common.persistence.repository;
 
-import app.coronawarn.server.common.persistence.domain.FederationBatch;
-import app.coronawarn.server.common.persistence.domain.FederationBatchStatus;
-import java.util.Date;
+import app.coronawarn.server.common.persistence.domain.FederationBatchInfo;
+import java.time.LocalDate;
+import java.util.List;
 import org.springframework.data.jdbc.repository.query.Modifying;
 import org.springframework.data.jdbc.repository.query.Query;
 import org.springframework.data.repository.PagingAndSortingRepository;
 import org.springframework.data.repository.query.Param;
 import org.springframework.stereotype.Repository;
 
 @Repository
-public interface FederationBatchRepository extends PagingAndSortingRepository<FederationBatch, Long> {
+public interface FederationBatchInfoRepository extends PagingAndSortingRepository<FederationBatchInfo, String> {
 
   @Modifying
-  @Query("INSERT INTO federation_batch "
+  @Query("INSERT INTO federation_batch_info "
       + "(batch_tag, date, status) "
       + "VALUES (:batchTag, :date, :status) "
       + "ON CONFLICT DO NOTHING")
   void saveDoNothingOnConflict(
       @Param("batchTag") String batchTag,
-      @Param("date") Date date,
-      @Param("status") FederationBatchStatus status
-  );
+      @Param("date") LocalDate date,
+      @Param("status") String status);
 
-  @Query("SELECT * FROM federation_batch WHERE status != 'PROCESSED' OR status is null ORDER BY date asc LIMIT 1")
-  FederationBatch findOldestUnprocessedFederationBatch();
+  @Modifying
+  @Query("INSERT INTO federation_batch_info "
+      + "(batch_tag, date, status) "
+      + "VALUES (:batchTag, :date, :status) "
+      + "ON CONFLICT (batch_tag) DO UPDATE SET status=:status")
+  void saveDoUpdateStatusOnConflict(
```

Comment:
do we need both `saveDoUpdateStatusOnConflict` vs. `saveDoNothingOnConflict` ?
isn't `saveDoUpdateStatusOnConflict` sufficient?

Reply:
Yes, we need to ensure "DoNothingOnConflict", when new BatchInfos are stored coming in as nextBatchTag (or future callback). In this case, there might come in duplicates of batchTags, which should be ignored. On the other hand, we also need a way to update a status, which is the reason for "UpdateStatusOnConflict"